plugins/user.py

```diff
@@ -0,0 +1,507 @@
+
+    @respond_to("^twofactor remove (?P<nick>\w+)")
+    @warn_direct_message
+    @requires_permission('administer_twofactor')
+    def remove_user_twofactor(self, message, nick):
+        """twofactor remove [nick]: remove [nick]'s twofactor authentication (requires the 'administer_twofactor' permission)"""
+        user_to_remove = User.get_from_nick(self, nick)
+        if user_to_remove:
+            user_to_remove.delete()
+            self.direct_reply(message, "Successfully removed twofactor authentication for user '{}'".format(nick))
+        else:
+            self.direct_reply(message, "I could not find user '{}', no action was taken".format(nick))
+
+    @respond_to("^show permissions for (?P<nick>\w+)")
+    @warn_direct_message
+    @requires_permission('view_permissions')
+    def show_user_permission(self, message, nick):
+        """show permissions for [nick]: get permissions for a user (requires the 'view_permissions' permission)"""
+        user = User.get_from_nick(self, nick)
+        if not user:
+            self.direct_reply(message, "I could not find user '{}'".format(nick))
+        else:
+            self.direct_reply(message, "User '{}' has permissions {}".format(user.nick, ', '.join(user.permissions)))
+
+    @respond_to("^who can (?P<permission>([\w.:-]+))")
+    @warn_direct_message
+    @requires_permission('view_permissions')
+    def get_users_for_permissions(self, message, permission):
+        """who can [permission]?: see which users have a particular permission (requires the 'view_permissions' permission)"""
+
+        nicks_with_perm = []
+        for user in User.list(self):
+            if user.has_permission(permission):
+                nicks_with_perm.append(user.nick)
+
+        if len(nicks_with_perm) > 0:
+            self.direct_reply(message, "The following users have permission to '{}': {}".format(permission, ', '.join(nicks_with_perm)))
+        else:
+            self.direct_reply(message, "No users have permission to '{}'".format(permission))
+
+    @respond_to("^can I(?P<permissions>( [\w.:-]+)+)")
+    @warn_direct_message
+    def confirm_user_permission(self, message, permissions):
+        """can I [permission]?: check if you have a specific permission"""
+        user = User.get_from_message(self, message)
+        if not user:
+            return
+
+        if not user.is_authenticated:
+            self.direct_reply(message, "I am unable to check your permissions because you are not authenticated. Please start an authenticated session by saying 'twofactor verify [token]'.")
+            return
+
+        for permission in permissions.split():
+            if user.has_permission(permission):
+                self.direct_reply(message, "Yes, you can {}".format(permission))
+            else:
+                self.direct_reply(message, "No, you can't {}".format(permission))
+
+    @respond_to("^give (?P<nick>\w+) permission(?P<permissions>( [\w.:-]+)+)")
```

Comment:
Heh, what do you think @fredsmith? I don't really have a strong preference here...